src/infra/src/table/templates.rs

```diff
@@ -0,0 +1,147 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use config::{ider, meta::alerts::templates::Template};
+use sea_orm::{
+    ActiveModelTrait, ColumnTrait, DatabaseConnection, EntityTrait, ModelTrait, QueryFilter, Set,
+    TryIntoModel,
+};
+
+use crate::{
+    db::{connect_to_orm, ORM_CLIENT},
+    errors,
+    table::{
+        entity::templates::{ActiveModel, Column, Entity, Model},
+        get_lock,
+    },
+};
+
+impl From<Model> for Template {
+    fn from(value: Model) -> Self {
+        Self {
+            name: value.name,
+            body: value.body,
+            is_default: value.is_default.then_some(true),
+            template_type: value.r#type.as_str().into(),
+            title: value.title.unwrap_or_default(),
+        }
+    }
+}
+
+pub async fn put(org_id: &str, template: Template) -> Result<Template, errors::Error> {
+    // make sure only one client is writing to the database(only for sqlite)
+    let _lock = get_lock().await;
+
+    let client = ORM_CLIENT.get_or_init(connect_to_orm).await;
+    let model: Model = match get_model(client, org_id, &template.name).await? {
+        Some(model) => {
+            let mut active: ActiveModel = model.into();
+            active.org = Set(org_id.to_string());
+            active.name = Set(template.name);
+            active.is_default = Set(template.is_default.unwrap_or_default());
+            active.r#type = Set(template.template_type.to_string());
+            active.body = Set(template.body);
+            active.title = Set((!template.title.is_empty()).then_some(template.title));
+            active.update(client).await?.try_into_model()?
+        }
+        None => {
+            let active = ActiveModel {
+                id: Set(ider::uuid()),
+                org: Set(org_id.to_string()),
+                name: Set(template.name),
+                is_default: Set(template.is_default.unwrap_or_default()),
+                r#type: Set(template.template_type.to_string()),
+                body: Set(template.body),
+                title: Set((!template.title.is_empty()).then_some(template.title)),
+            };
+            active.insert(client).await?.try_into_model()?
+        }
+    };
+    Ok(model.into())
+}
+
+pub async fn get(org_id: &str, name: &str) -> Result<Option<Template>, errors::Error> {
+    // make sure only one client is writing to the database(only for sqlite)
+    let _lock = get_lock().await;
```

Comment:
I think Prabhat and Hengfei mentioned in a Slack thread that we only need to worry about locking the Sqlite connection to avoid concurrent writes. My impression from the conversation was that this isn't necessary for reads.